libs/core/datetime/datetime-utils.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Converts a given time string to a desired format.
+ *
+ * @param {string} inputStr - The input time string to be converted.
+ * @returns {string} - The converted time string in the desired format.
+ */
+export function convertToDesiredFormat(inputStr: string): string {
+    const dayPeriodMapping = {
```

Comment:
Would it be possible to make this customizable? Perhaps the developer can pass in their own mapping object, something like

```
{
        morning: 4,
        afternoon: 12,
        evening: 16,
        night: 20
};
```

And there, anything between 20h and 4h would be considered "night"

Reply:
it's possible, if we receive similar requirement we will update convertToDesiredFormat method to just pass dayPeriodMapping and existing logic will handle it well.